consensus/beacon/consensus.go

```diff
@@ -257,7 +257,18 @@ func (beacon *Beacon) verifyHeader(chain consensus.ChainHeaderReader, header, pa
 		return consensus.ErrInvalidNumber
 	}
 	// Verify the header's EIP-1559 attributes.
-	return misc.VerifyEip1559Header(chain.Config(), parent, header)
+	if err := misc.VerifyEip1559Header(chain.Config(), parent, header); err != nil {
+		return err
+	}
+	// Verify existence / non-existence of withdrawalsHash.
+	shanghai := chain.Config().IsShanghai(new(big.Int).SetUint64(header.Time))
```

Comment:
Should use the `IsX(uint64)` api.

Reply:
bump ^

Reply:
Ah, never mind

Reply:
Needs to be rebased once https://github.com/ethereum/go-ethereum/pull/26474 is merged